lib/src/pubspec.dart

```diff
@@ -116,9 +116,30 @@ class Pubspec {
     }
   }
 
-  factory Pubspec.fromJson(Map json) => _$PubspecFromJson(json);
+  factory Pubspec.fromJson(Map json, {bool lenient = false}) {
+    lenient ??= false;
+
+    if (lenient) {
+      for (;;) {
```

Comment:
Please leave a comment that you try to parse, attempting to remove each top-level property that causes parsing errors...

---
On topic, what happens if you have an error nested somewhere inside the sub-tree...?

```dart
    test('bad repository url', () {
      final value = parse(
        {
          'name': 'foo',
          'dependencies': {'test': {'invalid': true}},
        },
        lenient: true,
      );
      expect(value.name, 'foo');
      expect(value.repository, isNull);
    });
```

Reply:
In that case the original exception is re-thrown (the retry-check checks for source map as well as existing key).

Reply:
Can we add a test case?

Reply:
Only to the `dependencies` or to the `environment` maps, because otherwise the `flutter` Map is not yet typed. Would that be covering it?

Reply:
Added a test with dependencies.